utils/java/mvn_test.go

```diff
@@ -224,8 +224,14 @@ func TestRunProjectsCmd(t *testing.T) {
 	_, cleanUp := coreTests.CreateTestWorkspace(t, filepath.Join("..", "..", "tests", "testdata", "maven-example"))
 	defer cleanUp()
 	mvnDepTreeManager := NewMavenDepTreeManager(&DepTreeParams{}, Projects, false)
-	output, err := mvnDepTreeManager.RunMavenDepTree()
+	output, clearMavenDepTreeRun, err := mvnDepTreeManager.RunMavenDepTree()
 	assert.NoError(t, err)
+	if assert.NotNil(t, clearMavenDepTreeRun) {
+		defer func() {
+			err = clearMavenDepTreeRun()
```

Comment:
why we need to defer? you can just apply, after that there are only checks on values you already have